Packages/com.chisel.editor/Chisel/Editor/UtilityWindows/SurfaceEditorWindow/SurfaceEditor.GUI.cs

```diff
@@ -0,0 +1,302 @@
+/*
+ * $TODO: clean up layout code - headers can be put into their own methods, etc.
+ */
+
+using Chisel.Components;
+using Chisel.Core;
+using UnityEditor;
+using UnityEngine;
+using EditorGUILayout = UnityEditor.EditorGUILayout;
+
+namespace Chisel.Editors
+{
+    internal partial class SurfaceEditor
+    {
+        private float           m_PanIncrementU   = 1.0f;
+        private float           m_PanIncrementV   = 1.0f;
+        private float           m_RotIncrement    = 15.0f;
+        private float           m_ScaleIncrementU = 1.0f;
+        private float           m_ScaleIncrementV = 1.0f;
+        private int             m_CurrentTab      = 0;
+        private LayerUsageFlags m_LayerUsage      = LayerUsageFlags.All;
+
+        // $TODO: find a way to edit surface flags outside of model... its serialized by unity and not visible to SurfaceReference or ChiselBrushMaterial.
+        // ^ this can be done, reference new EditorTools code for this
+        //private SurfaceFlags    m_WorldSpaceTex = SurfaceFlags.TextureIsInWorldSpace;
+
+        private void DrawTabBar()
+        {
+            GUILayout.BeginHorizontal();
+
+            m_CurrentTab = GUILayout.Toolbar
+            (
+                m_CurrentTab, new string[] { "Flags", "UVs", "Custom", "Other" }, m_TabStyle
+            );
```

Comment:
I'm wondering about this, if we want to make tabs part of our ux then it might make sense to have like a wrapper around the whole tabs thing, so you only give GUIContents (titles) and methods to fill in the contents of a tab or something.